pkg/sql/create_table.go

```diff
@@ -1698,6 +1698,140 @@ func makeTableDesc(
 			createStmt = &newCreateStmt
 		}
 	}
+	// First, process "CREATE TABLE LIKE" statements by moving the definitions from
+	// the referenced table into the AST being processed.
+	var newDefs tree.TableDefs
+	for i, def := range n.Defs {
+		d, ok := def.(*tree.LikeTableDef)
+		if !ok {
+			if newDefs != nil {
+				newDefs = append(newDefs, def)
+			}
+			continue
+		}
+		// We're definitely going to be editing n.Defs now, so make a copy of it.
+		if newDefs == nil {
+			newDefs = make(tree.TableDefs, 0, len(n.Defs))
+			newDefs = append(newDefs, n.Defs[:i]...)
+		}
+		td, err := ResolveMutableExistingObject(params.ctx, params.p, &d.Name, true /*required*/, ResolveRequireTableDesc)
+		if err != nil {
+			return ret, err
+		}
+		opts := tree.LikeTableOpt(0)
+		// Process ons / offs.
+		for _, opt := range d.Options {
+			if opt.Excluded {
+				opts &^= opt.Opt
+			} else {
+				opts |= opt.Opt
+			}
+		}
+
+		defs := make(tree.TableDefs, 0)
+		// Add all columns. Columns are always added.
+		for i := range td.Columns {
+			c := &td.Columns[i]
+			if c.Hidden {
+				// Hidden columns automatically get added by the system; we don't need
+				// to add them ourselves here.
+				continue
+			}
+			def := tree.ColumnTableDef{
```

Comment:
I think this should be pulled out into a function taking a column descriptor and spitting out a column table def, given some options